src/postgres/tree/PostgresFunctionsTreeItem.ts

```diff
@@ -32,13 +36,63 @@ export class PostgresFunctionsTreeItem extends AzureParentTreeItem<ISubscription
     }
 
     public async loadMoreChildrenImpl(): Promise<PostgresFunctionTreeItem[]> {
+        const rows: IPostgresProceduresQueryRow[] = await this.listFunctions();
+        this._functionsAndSchemas = {};
+        for (const row of rows) {
+            if (this._functionsAndSchemas[row.name]) {
+                this._functionsAndSchemas[row.name].push(row.schema);
+            } else {
+                this._functionsAndSchemas[row.name] = [row.schema];
+            }
+        }
+
+        return rows.map(row => new PostgresFunctionTreeItem(
+            this,
+            row,
+            this._functionsAndSchemas[row.name].length > 1
+        ));
+    }
+
+    public async createChildImpl(_context: IActionContext): Promise<PostgresFunctionTreeItem> {
+        const schemas: string[] = await this.listSchemas();
+        let schema: string;
+
+        if (schemas.length === 1) {
+            schema = schemas[0];
+        } else {
+            const schemaQuickPicks = schemas.map(s => { return { label: s }; });
+            schema = (await ext.ui.showQuickPick(schemaQuickPicks, {
+                placeHolder: localize('selectSchema', 'Select schema for new function')
+            })).label;
+        }
+
+        const name: string = (await ext.ui.showInputBox({
+            prompt: localize('provideFunctionName', 'Provide function name'),
+            validateInput: value => this.validateFunctionName(value, schema)
+        })).trim();
+
+        if (this._functionsAndSchemas[name]) {
+            this._functionsAndSchemas[name].push(schema);
+        } else {
+            this._functionsAndSchemas[name] = [schema];
+        }
+
+        const definition: string = defaultFunctionDefinition(schema, name);
```

Comment:
Call this before you add to the _functionsAndSchemas array.  That way if the query errors out, you won't have pushed in the schema.